### ISSUE TITLE
Snyk Report - (decompress-tar@4.1.1), Arbitrary File Write via Archive Extraction (Zip Slip)

### ISSUE THREAD

**Issue by BrettFieber** [serverless-operations/serverless-step-functions#588]
# This is a (Bug Report / Feature Proposal)

## Description

Snyk (https://snyk.io/) is reporting a security issue with serverless-step-functions@3.15.0 due to a dependency on decompress@4.2.1 => decompress-tar@4.1.1

https://security.snyk.io/vuln/SNYK-JS-DECOMPRESSTAR-559095

## Additional Data
* ***Serverless Framework Core Version you're using***: 2.72.4
* ***The Plugin Version you're using***: 3.15.0
* ***Operating System***: windows/linux

**Comment by niallbroadley** [comment]
This seems like the only package that we could have any hope of getting this resolved in:

- [@serverless/utils](https://www.npmjs.com/package/@serverless/utils?activeTab=readme) - last publish 3 years ago (github repo is [gone](https://github.com/serverless/utils), can't find any info on what's happened to this)
- [decompress](https://www.npmjs.com/package/decompress) - last publish 6 years ago
- [decompress-tar](https://www.npmjs.com/package/decompress-tar) - last publish 9 years ago

```
└─┬ serverless-step-functions@3.29.0
  └─┬ @serverless/utils@6.15.0
    └─┬ decompress@4.2.1
      ├── decompress-tar@4.1.1
      ├─┬ decompress-tarbz2@4.1.1
      │ └── decompress-tar@4.1.1 deduped
      └─┬ decompress-targz@4.1.1
        └── decompress-tar@4.1.1 deduped
```

It might be safe to assume `@serverless/utils` will never be upgraded so is a fix here possible?

**Comment by VirtueMe** [comment]
Good catch @niallbroadley — you're right that `@serverless/utils` is a direct dependency of this plugin, not just the Framework.

We audited the codebase and confirmed the package is not imported anywhere in `lib/`. It's a dead dependency. We're removing it in #753, which will also drop the `decompress-tar` chain from the dependency tree entirely and resolve this vulnerability.

**Comment by niallbroadley** [comment]
Thanks for this. Wondering when a fix might make its way into release as I see there has been one since the code was merged to master in [#753](https://github.com/serverless-operations/serverless-step-functions/pull/753)/[#754](https://github.com/serverless-operations/serverless-step-functions/pull/754)?

**Comment by VirtueMe** [comment]
Yes — it's already shipped. 3.29.2 was released on 2026-05-04 and includes #754. You can verify the dependency tree is clean:

```
$ npm view serverless-step-functions@3.29.2 dependencies
{
  '@osls/compose': '^1.4.0',
  'asl-validator': '^3.11.0',
  bluebird: '^3.4.0',
  chalk: '^4.1.2',
  joi: '^17.7.0',
  lodash: '^4.17.11'
}
```

`@serverless/utils` (and with it the `decompress-tar` chain) is no longer pulled in. Upgrading to 3.29.2 should resolve the advisory for you.

**Comment by niallbroadley** [comment]
Excellent. I had to `npm update @osls/compose` also as I had it set to 1.4.0 in my lock file and that version brings in `@serverless/utils` itself.

Thanks again.
